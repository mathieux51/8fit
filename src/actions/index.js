export const setField = field => ({
  type: "SET_FIELD",
  field
})
