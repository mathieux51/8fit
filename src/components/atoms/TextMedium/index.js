import React from "react"
import { Text } from "react-native"
import PropTypes from "prop-types"
import styled from "styled-components"

const StyledText = styled(Text)`
  font-size: ${({ theme }) => theme.medium};
  font-family: "fira-sans-medium";
`

const TextMedium = props => (
  <StyledText style={props.style}>{props.children}</StyledText>
)

TextMedium.propTypes = {
  style: PropTypes.array,
  children: PropTypes.string.isRequired
}

export default TextMedium
