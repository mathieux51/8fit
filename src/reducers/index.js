import { combineReducers } from "redux"
import nav from "./nav"
import user from "./user"

const rootReducer = combineReducers({
  nav,
  user
})

export default rootReducer
