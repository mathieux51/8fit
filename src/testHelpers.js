import React from "react"
import { ThemeProvider } from "styled-components"
import theme from "src/styles/theme"

export function wrappeWithThemeProvider(node) {
  return <ThemeProvider theme={theme}>{node}</ThemeProvider>
}
