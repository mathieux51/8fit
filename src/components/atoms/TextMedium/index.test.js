import React from "react"
import TextMedium from "./"
import { shallow } from "enzyme"
import { wrappeWithThemeProvider } from "src/testHelpers"

describe("<TextMedium />", () => {
  it("renders correctly", () => {
    expect(
      shallow(wrappeWithThemeProvider(<TextMedium>test</TextMedium>))
    ).toMatchSnapshot()
  })
})
