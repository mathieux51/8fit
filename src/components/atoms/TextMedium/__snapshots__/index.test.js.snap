// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextMedium /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
      Object {
        "black": "#292C2F",
        "bold": 20,
        "boldBig": 24,
        "boldBigger": 32,
        "green": "#26D88C",
        "grey": "#959798",
        "lGrey": "#EAEAEA",
        "medium": 12,
        "regular": 16,
        "regularSmall": 12,
        "white": "#ffffff",
      }
    }
  >
    <TextMedium>
      test
    </TextMedium>
  </ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "test",
    },
    "ref": null,
    "rendered": "test",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
